Update documentation with new tutorials and modernized installation guide (#20)
RxnRover/RxnRover: docs/source/dev_resources/tutorials/new_vipm_package/index.rst

Reviewer: Indentation is slightly off on many of the bullets in this file. Mostly, it is sub-bullets under a numbered bullet, which should be three spaces instead of two, I think. I pointed out the first few places so you can see what I mean, but please check the rest of the document since many of them had the issue and I just didn't suggest the fix for each one.

Author: Okay doke, I went ahead and adjusted the indentation in the other sections.

.. _create-new-vipm-package:

Create a VIPM Package
=====================

The `Virtual Package Manager (VIPM) <https://www.vipm.io/>`_ is a tool for managing LabVIEW packages, including instrument drivers and plugins. Using VIPM, you can create a package from your LabVIEW project that can be easily shared and installed by others.

This tutorial will guide you through the process of creating a new VIPM package for use with the Rxn Rover platform.

General Steps to Create a VIPM Package
--------------------------------------

1. **Open VIPM**  
   Launch the VI Package Manager application.

2. **Create a New Package**  

   - Click the **Open VI Package Builder** icon in the VIPM toolbar (top-left).
   - In the Package Builder menu, select **Create New**.
   - Set the **Source Directory** to the root folder of your LabVIEW project.
   - Set the **Product Name** to a meaningful name (e.g., `"IR Sensor Plugin"`).
   - Set the **Company Name** to:

>     - Rxn Rover if you're a first-party developer.
>     - Your own company name if you're a third-party developer.

Reviewer: Indentation is a slightly off here.
```suggestion
     - Rxn Rover if you're a first-party developer.
     - Your own company name if you're a third-party developer.
```

Author: Made this change

   This will create a new ``.vipb`` (VI Package Builder) file named after the Product Name in your project directory.

3. **Configure Basic Build Information**  

   - **Source Directory**: Points to your LabVIEW project root.
   - **Build Output Directory**: Choose a location like ``<your-workspace>\VIPMPackages``.

.. note::
   The rest of the configuration differs slightly depending on whether you're a **first-party** (Rxn Rover) or **third-party** developer.

First-Party Plugins (Rxn Rover Developers)
------------------------------------------

These instructions are for Rxn Rover developers creating official plugins or drivers.

1. **Display Information**

  - **Product Name**: Name of your plugin (e.g., "IR Sensor Plugin").
  - **Icon**: Optional.
  - **Company Name**: Rxn Rover
  - **Author Name**: Rxn Rover
  - **Product Homepage**: ``https://github.com/RxnRover/<your-plugin-repo>``

>   - **Copyright**: Copyright (c) `<year>` RxnRover

Reviewer: ```suggestion
  - **Copyright**: Copyright (c) ``<year>`` RxnRover
```

>   - **Product Name**: Name of your plugin (e.g., "IR Sensor Plugin").
>   - **Icon**: Optional.
>   - **Company Name**: Rxn Rover
>   - **Author Name**: Rxn Rover
>   - **Product Homepage**: ``https://github.com/RxnRover/<your-plugin-repo>``
>   - **Copyright**: Copyright (c) `<year>` RxnRover
>   - **License Agreement**: Path to ``LICENSE.txt`` in your project folder.
>   - **Summary**: Short description (e.g., "A plugin for IR sensor data acquisition and processing").

Reviewer: Indentation is slightly off I think.
```suggestion
   - **Product Name**: Name of your plugin (e.g., "IR Sensor Plugin").
   - **Icon**: Optional.
   - **Company Name**: Rxn Rover
   - **Author Name**: Rxn Rover
   - **Product Homepage**: ``https://github.com/RxnRover/<your-plugin-repo>``
   - **Copyright**: Copyright (c) `<year>` RxnRover
   - **License Agreement**: Path to ``LICENSE.txt`` in your project folder.
   - **Summary**: Short description (e.g., "A plugin for IR sensor data acquisition and processing").
```

2. **Destinations**

>   - Set install location to:
>     - ``Documents\RxnRover\Plugins\<plugin-type>\<Product Name>`` for plugins.
>     - ``instr.lib\RxnRover\<Product Name>`` for drivers.
> 
>   - **Package Filename**:  
>     Use this format:  
>     ``rxn_rover_lib_<product_name>-X.X.X.X.vip``

Reviewer: ```suggestion
   - Set install location to:
     - ``Documents\RxnRover\Plugins\<plugin-type>\<Product Name>`` for plugins.
     - ``instr.lib\RxnRover\<Product Name>`` for drivers.

   - **Package Filename**:  
     Use this format:  
    ``rxn_rover_lib_<product_name>-X.X.X.X.vip``
```

Continue with the rest of the tutorial at :ref:`build-and-share-package`.

Third-Party Plugins
-------------------

For external developers who wish to make their plugin compatible with Rxn Rover.

1. **Display Information**

  - **Product Name**: Name of your plugin (e.g., "IR Sensor Plugin").
  - **Icon**: Optional.

>   - **Company Name**: `Your organization's name`
>   - **Author Name**: `Your personal or your organization's name`

Reviewer: ```suggestion
  - **Company Name**: *Your organization's name*
  - **Author Name**: *Your personal or your organization's name*
```

  - **Product Homepage**: e.g., ``https://github.com/<github_username>/<your_plugin_repo.>``

>   - **Copyright**: Copyright (c) `<year>` `<Your Company>`

Reviewer: ```suggestion
  - **Copyright**: Copyright (c) ``<year>`` ``<Your Company>``
```

  - **License Agreement**: Path to ``LICENSE.txt`` in your project folder.
  - **Summary**: A brief description of the plugin.
  - **Description**: Provide a more detailed explanation of the plugin’s features and how to use it.

2. **Destinations**

  - For plugins:  
    ``Documents\RxnRover\Plugins\<plugin-type>\<Product Name>``

>   - For drivers:  
>     ``instr.lib\RxnRover\<Product Name>``

Reviewer: Most people actually seem to put their drivers straight into `instr.lib\<Product Name>`. The `instr.lib\RxnRover` subfolder is actually just a convention I use to make ours easy to find. This should be changed to either `instr.lib\<Product Name>`, `instr.lib\<Company Name>\<Product Name>`, or both.
```suggestion
  - For drivers:  
    ``instr.lib\RxnRover\<Product Name>``
```

.. note::

>    Keep install paths consistent with Rxn Rover’s plugin architecture to ensure compatibility.

Reviewer: ```suggestion
   Keep install paths consistent with Rxn Rover's plugin architecture to ensure compatibility.
```

3. **Source File Settings**

  - Include:
    - All required ``.vi`` files
    - Configuration files
    - Documentation (e.g., ``README.md``, ``LICENSE.txt``)

  - Exclude: ``.git``, ``.gitignore``, ``.lvproj``, other unnecessary dev files.

  - Set the **Package Source Destination** to the install path you defined in the **Destinations** tab.
  - This applies to all subfolders unless explicitly overridden.

 4. **Package Dependencies**

  - VIPM will auto-detect most dependencies.
  - Manually add:
  
    - `RxnRover <https://rxnrover.github.io/PluginCatalog/core_tools/software/rxn_rover.html>`_ (core library)
    - `Dynamic Reentrant <https://rxnrover.github.io/PluginCatalog/core_tools/libraries/dynamic_reentrant.html>`_ (required for plugin loading)
    - Any other required plugins or drivers.

.. important::

>    If your plugin depends on a specific driver—or vice versa—be sure to include that in the dependencies.

Reviewer: I don't know that the reverse will happen where a driver depends on a specific plugin, or at least I can't think of a likely scenario for this, so I'd recommend removing that part.
```suggestion
   If your plugin depends on a specific driver, be sure to include that in the dependencies.
```

Author: Makes sense, I'll go ahead and make this change

5. **Install Requirements**

  - Set compatible:

    - **Operating Systems**
    - **LabVIEW Versions**

.. _build-and-share-package:

Build and Share the Package
---------------------------

> 1. Once configuration is complete, click :guilabel:`Build Package(Arrow)` in the VIPM toolbar.

Reviewer: I'm not sure what "Build Package(Arrow)" means here. Does the Build Package button have an arrow glyph on it? (I'm not on a system to easily open and check VIPM right now) If so, since they may change it from an arrow in the future, I'd say just remove the `(Arrow)` part:
```suggestion
1. Once configuration is complete, click :guilabel:`Build Package` in the VIPM toolbar.
```
If you want to keep the descriptor, which would be useful for users as long as it doesn't change, it may be clearer to specify the appearance afterward. Something like this (`<your_version>` because I don't know your version of VIPM):
```suggestion
1. Once configuration is complete, click :guilabel:`Build Package` in the VIPM toolbar (an arrow symbol in VIPM v`<your_version>`).
```

Author: Yea, this was a little weird, The build package button is actually an arrow, I'll just get rid of the (arrow)

1. **Display Information**

  - **Product Name**: Name of your plugin (e.g., "IR Sensor Plugin").
  - **Icon**: Optional.
  - **Company Name**: `RxnRover`
  - **Author Name**: `RxnRover`
  - **Product Homepage**: ``https://github.com/RxnRover/<your-plugin-repo>``
  - **Copyright**: `Copyright (c) <year> RxnRover`
  - **License Agreement**: Path to ``LICENSE.txt`` in your project folder.
  - **Summary**: Short description (e.g., "A plugin for IR sensor data acquisition and processing").

2. **Destinations**

  - Set install location to:
    - ``Documents\RxnRover\Plugins\<plugin-type>\<Product Name>`` for plugins.
    - ``instr.lib\RxnRover\<Product Name>`` for drivers.

  - **Package Filename**:  
    Use this format:  
    ``rxn_rover_lib_<product_name>-X.X.X.X.vip``

> Continue with the rest of the tutorial at Build and Share the Package

Author: That's true, I must've forgotten to include that.

1. **Display Information**

>   - **Product Name**: e.g., `"IR Sensor Plugin"`

Reviewer: ```suggestion
  - **Product Name**: e.g., ``"IR Sensor Plugin"``
```
I could see this being italicized (as you had it), the back ticks removed altogether since it is quoted, or the inline code style (double back ticks). I just wanted to point it out to verify.

Author: Yea I meant to italicize it here, in general I wanted to distinguish visually between the more abstract _This is your own name_ rather than a literal. I appreciate you pointing it out though because on second-thought it seems there's a few situations where it should or shouldn't be italicized.

  - **Icon**: Optional.
  - **Company Name**: Your organization.
  - **Author Name**: You or your organization.
  - **Product Homepage**: e.g., your GitHub repo.

>   - **Copyright**: `Copyright (c) <year> <Your Company>`

Reviewer: ```suggestion
  - **Copyright**: ``Copyright (c) <year> <Your Company>``
```

Again just pointing this out to make sure the italicizing is what you wanted.

  - **License Agreement**: Path to ``LICENSE.txt`` in your project folder.
  - **Summary**: A brief description of the plugin.
  - **Description**: Provide a more detailed explanation of the plugin’s features and how to use it.

2. **Destinations**

  - For plugins:  
    ``Documents\RxnRover\Plugins\<plugin-type>\<Product Name>``

  - For drivers:  
    ``instr.lib\RxnRover\<Product Name>``

.. note::
   Keep install paths consistent with Rxn Rover’s plugin architecture to ensure compatibility.

3. **Source File Settings**

  - Include:
    - All required ``.vi`` files
    - Configuration files
    - Documentation (e.g., ``README.md``, ``LICENSE.txt``)

  - Exclude:
    - ``.git``, ``.gitignore``, ``.lvproj``, other unnecessary dev files.

  - Set the **Package Source Destination** to the install path you defined in the **Destinations** tab.
  - This applies to all subfolders unless explicitly overridden.

 4. **Package Dependencies**

  - VIPM will auto-detect most dependencies.
  - Manually add:
  

>     - `RxnRover` (core library)
>     - `Dynamic Reentrant` (required for plugin loading)

Reviewer: ```suggestion
    - ``RxnRover`` (core library)
    - ``Dynamic Reentrant`` (required for plugin loading)
```

Author: I actually think these were meant to be links to the plugin catalog once I had made them.  I don't necessarily want to use `inline literals` since it doesn't represent a piece of code,command, or file/directory. And for developer's who would be reading it I don't want to confuse them to much.

Reviewer: Got it, that makes sense. I see that it was changed to have the links in the latest version, so I am going to resolve this.
